### PR TITLE
Editor themes

### PR DESCRIPTION
**Pull request by szymski**
Ready for testing. Supports only TextMate themes, but I'm planning to make it possible to edit color values from the menu, like before. You can take themes from [Colorsublime](http://colorsublime.com).
[Preview](http://url.szymekk.me/bc032)

### CODE REVIEW

File: lua/starfall/editor/editor.lua [hunk 1]
```diff
@@ -831,6 +834,9 @@ if CLIENT then
 		SF.Editor.open ()
 		print("Editor reloaded")
 	end )
+	concommand.Add("sf_editor_reload", function()
```
Comment (Damianu) [inline lua/starfall/editor/editor.lua]: Change that to sf_editor_reloadfiles so its not confusing with previous command

File: lua/starfall/editor/editor.lua [hunk 2]
```diff
@@ -819,7 +822,7 @@ if CLIENT then
 				end
 			end
 		end )
-	concommand.Add( "sf_editor_reload", function()
+	concommand.Add( "sf_editor_restart", function()
```
Comment (Damianu) [inline lua/starfall/editor/editor.lua]: sf_editor_recreate would be more accurate, should name it like that myself.

File: lua/starfall/editor/syntaxmodes/starfall.lua [hunk 3]
```diff
@@ -52,30 +52,14 @@ local directives = {
...
-	return colors["notfound"][1]
-end
+function EDITOR:LoadSyntaxColors()
+	colors = { }
 
-function EDITOR:SetSyntaxColor( colorname, colr )
-	if not colors[colorname] then return end
-	colors[colorname][1] = colr
+	for k, v in pairs(SF.Editor.Themes.CurrentTheme) do
+		colors[k] = { v, false }
```
Comment (Damianu) [inline lua/starfall/editor/syntaxmodes/starfall.lua]: It should be v,bold - some themes may implement bold fonts or italic(didnt add that yet), but that false means "not bold"

File: lua/starfall/editor/editor.lua (hunk 2 above)
Comment (thegrb93) [inline lua/starfall/editor/editor.lua]: nobody uses recreate though. reload is the conventional 'recreate' term.

**Comment by thegrb93** [comment]
Crikey this is a big one. I'll test it now.

**Comment by thegrb93** [comment]
Yeah it looks fine. I'll let you guys work out the bold/italic issue in the next commit.

**Pull request merged**